provider/resources.go

```diff
@@ -454,14 +455,72 @@ func preConfigureCallbackWithLogger(credentialsValidationRun *atomic.Bool, gcpCl
 //go:embed cmd/pulumi-resource-gcp/bridge-metadata.json
 var metadata []byte
 
+// A predicate function that always returns true.
+func always[T any](T) bool { return true }
+
+// fixEmptyLabels applies a state edit to fix
+// https://github.com/pulumi/pulumi-gcp/issues/2372.
+func fixEmptyLabels(_ context.Context, req shimv2.PlanStateEdit) (cty.Value, error) {
+	inputs, ok := resource.PropertyPath{"labels"}.Get(resource.NewProperty(req.NewInputs))
+	if !ok {
+		return req.PlanState, nil
+	}
+
+	labels := resource.FromResourcePropertyValue(inputs)
+	if !labels.IsMap() {
+		// No labels found, so we don't need to correct
+		return req.PlanState, nil
+	}
+
+	// fixOutputLabels fixes falsely computed values inherited from "labels".
```

Comment:
Do you mean "unknown" not "computed"? I believe this are all computed either way.

Reply:
I meant computed from the Pulumi namespace, which is equivalent to the TF concept of unknown.